src/components/Popover.vue

```diff
@@ -0,0 +1,289 @@
+<template>
+  <div :class="{ 'popover': true, 'popover--active': isActive }" :style="elementStyle">
+    <slot></slot>
+  </div>
+</template>
+<script lang="ts">
+import _ from 'lodash';
+import Vue from 'vue';
+import { Component, Prop, Watch } from 'vue-property-decorator';
+import { POPOVER_ALIGN, POPOVER_SHADOW_GAP } from '@/lib/popover';
+
+/**
+ * We use weak typing as it is used to define CSS rules
+ * And that we're not forced to define all four properties
+ */
+interface ElementPosition {
+  top?: string | number;
+  bottom?: string | number;
+  left?: string | number;
+  right?: string | number;
+}
+
+/**
+@type {VueComponent}
+@name Popover
+
+@description
+Implement a popover component
+
+@params {Boolean} [active = false] toggles visibility
+@params {String} [align = 'center'] sets the popover's default alignment:
+                                    - possible values work like the text-align
+                                      CSS property
+                                    - justify adapts the popover to its parent
+                                      (same width and horizontal position)
+@params {Boolean} [bottom = false] sets the default position to below instead of
+                                   above
+*/
+@Component({
+  name: 'popover',
+})
+export default class Popover extends Vue {
+  // To specify TypeScript that we got a HTMLElement
+  $el!: HTMLElement;
+
+  @Prop({
+    type: Boolean,
+    default: () => false,
+  })
+  active!: boolean;
+
+  @Prop({
+    default: () => POPOVER_ALIGN.CENTER,
+    validator: value => _.includes(POPOVER_ALIGN, value),
+  })
+  align!: string;
+
+  @Prop({
+    type: Boolean,
+    default: () => false,
+  })
+  bottom!: boolean;
+
+  elementStyle: ElementPosition = {};
+  parent: HTMLElement | null = null;
+  parents: Array<HTMLElement> = [];
+
+  get alignMethod() {
+    switch (this.align) {
+      case POPOVER_ALIGN.CENTER:
+        return this.alignCenter;
+      case POPOVER_ALIGN.JUSTIFY:
+        return this.alignJustify;
+      case POPOVER_ALIGN.LEFT:
+        return this.alignLeft;
+      case POPOVER_ALIGN.RIGHT:
+        return this.alignRight;
+      default:
+        return () => ({});
+    }
+  }
+
+  get isActive() {
+    return this.active;
+  }
+
+  get isBottom() {
+    return this.bottom;
+  }
+
+  mounted() {
+    // Save original parent before moving into body to use its position
+    this.parent = this.$el.parentElement;
+
+    // Move the popover into the body to prevent its hiding by an
+    // `overflow: hidden` container
+    document.body.appendChild(this.$el);
+
+    // Get all scrollable parents
+    const parents = [];
+    let { parent } = this;
+
+    while (parent !== document.body && !_.isNull(parent)) {
```

Comment:
Any reason not to do:
```suggestion
    while (parent !== document.body && parent !== null) {
```